src/hotspot/share/gc/shared/gc_globals.hpp

```diff
@@ -693,8 +693,12 @@
   product(uint, GCCardSizeInBytes, 512,                                     \
           "Card table entry size (in bytes) for card based collectors")     \
           range(128, NOT_LP64(512) LP64_ONLY(1024))                         \
-          constraint(GCCardSizeInBytesConstraintFunc,AtParse)
-  // end of GC_FLAGS
+          constraint(GCCardSizeInBytesConstraintFunc,AtParse)               \
+                                                                            \
+  product(bool, UseAltGCForwarding, false, EXPERIMENTAL,                    \
+          "Use alternative GC forwarding that preserves object headers")    \
```

Comment:
I believe this feature should not be a product flag/available in product builds at this point.

Currently there is no upside for end users to make this a product flag of any kind, and changing this temporarily for Lilliput development seems a minor annoyance. Just like we do with other code that if we want it for product builds for testing like induced evacuation failure.

Making it a develop flag (and with a default value of 'false') the whole related code will (likely) be compiled out completely, having no performance impact guaranteed. >5% increased full gc pauses for no reason (for end users) for the Serial collector does not make it useful to them.

As soon as it is actually needed and there is a way to profit from it (i.e. as soon as Lilliput is added; I assume that the increased heap density will give back this loss by decreasing the amount of garbage collections) this should be made a product flag. Just not at this time.